### DISCUSSION TITLE
Infinite Loading when connect metamask

### DISCUSSION THREAD

**Discussion by shin101004** [WalletConnect/walletconnect-monorepo#4207] [Developer Support]
Hi ! I use `@web3modal/wagmi-react-native`.

I request connect metamask using open method.

after i confirm connect and return to my dapp. it shows infinte loading state.

i just call open method.

```javascript
import { useWeb3Modal, useWeb3ModalState } from '@web3modal/wagmi-react-native';

const useWeb3 = () => {

const { open } = useWeb3Modal();

return {
  onOpen: () => open({ view: 'Account' })
  }
}
```

[![Watch the video](https://img.youtube.com/vi/T-D1KVIuvjA/maxresdefault.jpg)](https://github.com/WalletConnect/walletconnect-monorepo/assets/61345090/86e479fd-2938-4e01-9681-dc5e33d1344d)

**Comment by glitch-txs** [comment]
how could you share a minimal reproducible example?

**Reply by shin101004** [reply]
Sry, i cant do it now

it will upload after 2q

**Comment by ghost** [comment]
did you find the answer?

**Reply by shin101004** [reply]
nope, i cant find the answer.

in particular, there was an issue with the metamask.

if you encounter the issue, please close the pop-up and retry connect wallet.

i found the app which uses walletconnect library
https://chat.blockscan.com/start
above app is web3 app from etherscan.

etherscan used walletconnet too. they have same issue.

**Reply by shin101004** [reply]
i tried to clear all localstorage keys from walletconnect and reissue keys. it didnt help.

i think it happens when key doesnt match with my app and wallet app. 

unfortunately, there is no logs from app when issue occurred.
